DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/compute_units/src/pipe_array.hpp

```diff
@@ -0,0 +1,43 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <CL/sycl.hpp>
+#include <utility>
+
+// Header locations and some DPC++ extensions changed between beta09 and beta10
+// Temporarily modify the code sample to accept either version
+#define BETA09 20200827
+#if __SYCL_COMPILER_VERSION <= BETA09
+  #include <CL/sycl/intel/fpga_extensions.hpp>
+  namespace INTEL = sycl::intel;  // Namespace alias for backward compatibility
+#else
+  #include <CL/sycl/INTEL/fpga_extensions.hpp>
+#endif
+
+
+#include "pipe_array_internal.hpp"
+
+template <class Id, typename BaseTy, size_t depth, size_t... dims>
+struct PipeArray {
+  PipeArray() = delete;
+
+  template <size_t... idxs>
+  struct StructId;
+
+  template <size_t... idxs>
+  struct VerifyIndices {
+    static_assert(sizeof...(idxs) == sizeof...(dims),
+                  "Indexing into a PipeArray requires as many indices as "
+                  "dimensions of the PipeArray.");
+    static_assert(VerifierDimLayer<dims...>::template VerifierIdxLayer<
+                      idxs...>::IsValid(),
+                  "Index out of bounds");
+    using VerifiedPipe =
+        cl::sycl::INTEL::pipe<StructId<idxs...>, BaseTy, depth>;
```

Comment:
This is the cause of the compile error. For beta09/10 compatibility, the code aliases the INTEL namespace (new to beta10) back to sycl::intel (from beta09). But here the INTEL namespace is fully scoped.

 Please change it to:
`INTEL::pipe<StructId<idxs...>, BaseTy, depth>;`

This compiles for me on beta09 after this change.

Reply:
Thank you for the fix!